CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs

```diff
@@ -15,264 +16,321 @@ public class GoGenerator : ILanguageGenerator<SnippetModel, OpenApiUrlTreeNode>
-                    var propertyAssignment = $"{indentManager.GetIndent()}\"{property.Name}\": ";
-                    WriteProperty(payloadSB, property.Value, null, indentManager, propertyAssignment, null, ",");
-                }
-                indentManager.Unindent();
-                payloadSB.AppendLine($"{indentManager.GetIndent()}}}");
+            else
+            {
+                builder.AppendLine($"{indentManager.GetIndent()}{requestBodyVarName} := graphmodels.New{codeGraph.Body.Name}()");
+                WriteCodePropertyObject(requestBodyVarName, builder, codeGraph.Body, indentManager);
             }
         }
-        private static void WritePropertyFromTempVar(StringBuilder payloadSB, string propertyAssignment, string tempVarName, string valueAssignment, string valueDeclaration, bool addPointer, string propertySuffix) {
-            if(!string.IsNullOrEmpty(tempVarName)) {
-                payloadSB.AppendLine($"{tempVarName}{valueAssignment}{valueDeclaration}");
-                var pointer = addPointer ? "&" : string.Empty;
-                payloadSB.AppendLine($"{propertyAssignment}{pointer}{tempVarName}{propertySuffix}");
-            } else {
-                payloadSB.AppendLine($"{propertyAssignment}{valueDeclaration}{propertySuffix}");
+        private static string GetActionParametersList(params string[] parameters)
+        {
+            var nonEmptyParameters = parameters.Where(p => !string.IsNullOrEmpty(p));
+            if (nonEmptyParameters.Any())
+                return string.Join(", ", nonEmptyParameters.Aggregate((a, b) => $"{a}, {b}"));
+            else return string.Empty;
+        }
+
+        private static void WriteArrayProperty(string propertyAssignment, string objectName, StringBuilder builder, CodeProperty parentProperty, CodeProperty codeProperty, IndentManager indentManager){
+            var contentBuilder = new StringBuilder();
+            var propertyName = NormalizeJsonName(codeProperty.Name.ToFirstCharacterLowerCase());
+
+            var objectBuilder = new StringBuilder();
+            IndentManager indentManagerObjects = new IndentManager();
+
+            indentManager.Indent();
+            var childPosition = 0;
+            foreach (var child in codeProperty.Children){
+                if(child.PropertyType == PropertyType.Object){
+                    WriteCodeProperty(propertyAssignment, objectBuilder, codeProperty, child , indentManagerObjects, childPosition);
+                    contentBuilder.AppendLine($"{indentManager.GetIndent()}{child.Name.ToFirstCharacterLowerCase()?.IndexSuffix(childPosition)},");
+                }else{
+                    WriteCodeProperty(propertyAssignment, contentBuilder, codeProperty, child , indentManager, childPosition);
+                }
+                childPosition++;
             }
+            indentManager.Unindent();
+
+            if(objectBuilder.Length > 0){
+                builder.AppendLine("\n");
```

Comment:
unless you want this to be normalized to the Unix-way all the time:
```suggestion
                builder.AppendLine(Environment.NewLine);
```

Reply:
I don't think this will be normalized to the Unix-way, because you are already calling AppendLine which should already respect the environment while appending new lines.